src/components/shared/AuthEmail.tsx

```diff
@@ -30,28 +34,50 @@ const useStyles = makeStyles((theme) => ({
     buttonRow: { display: 'flex', justifyContent: 'space-between' },
 }));
 
-const ForgotPassword = () => {
+interface AuthEmailProps {
+    data: AuthEmailDataProps;
+    error?: HttpError;
+}
+
+interface AuthEmailDataProps {
+    history: boolean;
+    title: string;
+    description?: string;
+    button: string;
+}
+
+export const AuthEmail = ({ data, error }: AuthEmailProps) => {
     const history = useHistory();
+    const dispatch = useDispatch();
     const classes = useStyles();
+    const [email, setEmail] = useState<string>('');
+    const [openModal, setOpenModal] = useState(false);
+
+    function handleClick() {
+        dispatch(sendAuthEmail(email));
+        setOpenModal(true);
+    }
 
     return (
         <Container component="main" maxWidth="xs">
-            <CssBaseline />
+            <TransitionsModal open={openModal} setOpen={setOpenModal} />
             <div className={classes.paper}>
                 <Typography component="h1" variant="h5">
-                    Forgot Password
-                </Typography>
-                <Typography>
-                    Lost your password? Please enter your email address. You will receive
-                    a link to create a new password via email.
+                    {data.title}
                 </Typography>
+                {data.description ? <Typography>{data.description} </Typography> : null}
```

Comment:
instead of ternaries using null for all of these, can just do `{someBoolean && <MyComponent />}`

Reply:
ohh TIL